### PR TITLE
 handle noise with script

### PR DESCRIPTION
**Pull request by hamelsmu**
Related to https://github.com/fastai/imagenette/issues/40

This is mostly using @tmabraham's code with some small adjustments.   I tested this locally in a notebook after re-downloading the data I uploaded. 

cc: @jph00 @tmabraham

### CODE REVIEW

File: nbs/examples/train_imagenette.py [hunk 1]
```diff
@@ -13,23 +13,40 @@
...
-                       splitter=GrandparentSplitter(valid_name='val'),
-                       get_items=get_image_files, get_y=parent_label,
-                       item_tfms=[RandomResizedCrop(size, min_scale=0.35), FlipItem(0.5)],
-                       batch_tfms=batch_tfms)
-    return dblock.dataloaders(source, path=source, bs=bs, num_workers=workers)
+    
+    if pct_noise > 0:
+        csv_file = 'noisy_imagewoof.csv' if woof else 'noisy_imagenette.csv'
+        inp = pd.read_csv(source/csv_file)
+        dblock = DataBlock(blocks=blocks,
```
Comment (jph00) [inline nbs/examples/train_imagenette.py:30]: Could this be simplified by having a single `DataBlock` call, always using the CSV, and just using different columns for each noise level (including zero noise)?

File: nbs/examples/train_imagenette.py [hunk 2]
```diff
@@ -13,23 +13,40 @@
 def pr(s):
     if rank_distrib()==0: print(s)
 
-def get_dls(size, woof, bs, sh=0., workers=None):
+def get_dls(size, woof, pct_noise, bs, sh=0., workers=None):
+    assert pct_noise in [0,5,50], '`pct_noise` must be 0,5 or 50.'
```
Comment (jph00) [inline nbs/examples/train_imagenette.py:17]: This doesn't seem to match the list of noise levels in the help text

**Review by jph00 — CHANGES_REQUESTED** [review]
Thanks! Just some minor requests...

File: nbs/examples/train_imagenette.py (hunk 1 above)
Comment (hamelsmu) [reply nbs/examples/train_imagenette.py]: fixed

**Comment by hamelsmu** [comment]
@jph00 I updated this PR as requested.  Also, I checked in the new hashes that correspond to the updated tar files.  

@tmabraham to test, check out this branch and install it: from the root of this repo run `pip install -e .`

**Review by tmabraham — CHANGES_REQUESTED** [review]
`to_native_fp16` --> `to_fp16` (error with imagenette script)

File: nbs/examples/train_imagenette.py [hunk 3]
```diff
@@ -59,7 +68,7 @@ def main(
     elif opt=='sgd'   : opt_func = partial(SGD, mom=mom)
     elif opt=='ranger': opt_func = partial(ranger, mom=mom, sqr_mom=sqrmom, eps=eps, beta=beta)
 
-    dls = rank0_first(get_dls, size, woof, bs, sh=sh, workers=workers)
+    dls = rank0_first(get_dls, size, woof, pct_noise, bs, sh=sh, workers=workers)
     pr(f'epochs: {epochs}; lr: {lr}; size: {size}; sqrmom: {sqrmom}; mom: {mom}; eps: {eps}')
```
Comment (tmabraham) [inline nbs/examples/train_imagenette.py:72]:
```suggestion
    pr(f'pct_noise: {pct_noise}; epochs: {epochs}; lr: {lr}; size: {size}; sqrmom: {sqrmom}; mom: {mom}; eps: {eps}')
```
Printing out the amount of noise used at the beginning of the run

**Pull request merged**